form.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+/**
+ * Sets the change handler for the cluster select.
+ */
+function set_cluster_change_handler() {
+
+  // Ensure owens does not use spark 2.4.5 on page load.
+  toggle_options("batch_connect_session_context_spark_version");
+  // Ensure pitzer does not use jupyter_app 3.0 on page load.
+  toggle_options("batch_connect_session_context_version");
+
+  const cluster_input = $('#batch_connect_session_context_cluster');
+
+  cluster_input.change((event) => {
+    // spark version change.
+    toggle_options("batch_connect_session_context_spark_version");
+    // jupyter_app version change.
+    toggle_options("batch_connect_session_context_version");
+  });
+}
+
+function toggle_options(element_name) {
+  const cluster = current_cluster_capitalized();
+  const search = "#" + element_name + " option";
+  const options = $(search);
+
+  options.each(function(_i, option) {
+    // the variable 'option' is just a data structure. it has no attr, data, show
+    // or hide methods so we have to query for it again
+    let option_element = $(search + "[value='" + option.value + "']");
+    let data = option_element.data();
+    let show = data["optionFor" + cluster];
+
+    if (show == false) {
+      option_element.prop('disabled', true);
```

Comment:
I think we should hide, not disable. Disabling still shows the option - you just can't choose it.

I think hiding it has a better UX?

Reply:
I completely agree that hiding makes more sense for UX, otherwise we may get "bug" tickets that users can't select options. Made that change already and pushing it up.